web/src/components/auth/confirm-user/ConfirmUserForm.tsx

```diff
@@ -0,0 +1,142 @@
+'use client';
+
+import { AuthContext } from '@/context/AuthContext';
+import { TOO_MANY_REQUESTS, UNKNOWN } from '@/errors/ErrorMessages';
+import confirmUserSchema from '@/features/auth/confirm-user/schema';
+import { ConfirmUserRequest } from '@/hooks/useAuth';
+import { isAWSError } from '@/utils/awsUtils';
+import { USE_FORM_CONFIG } from '@/utils/forms';
+import { getItem } from '@/utils/localStorage';
+import {
+    Box,
+    Button,
+    FormGroup,
+    FormHelperText,
+    Stack,
+    TextField,
+    Typography,
+} from '@mui/material';
+import { useRouter } from 'next/navigation';
+import { useContext, useEffect, useState } from 'react';
+import { useForm } from 'react-hook-form';
+
+const DEFAULT_VALUES: ConfirmUserRequest = {
+    code: '',
+    username: '',
+};
+
+const useAlert = () => {
+    return (message: string, type: string) => {
+        console.log(message, type);
+    };
+};
+
+const ConfirmUserForm = () => {
+    const router = useRouter();
+    const showAlert = useAlert();
+    const [username, setUsername] = useState<string>('');
+    const { confirmUser } = useContext(AuthContext);
+    const { formState, register, handleSubmit } = useForm<ConfirmUserRequest>(
+        USE_FORM_CONFIG<ConfirmUserRequest>(DEFAULT_VALUES, confirmUserSchema),
+    );
+    const { isValid, isSubmitting, isDirty, errors } = formState;
+
+    useEffect(() => {
+        setUsername(getItem('username') ?? '');
+    }, []);
+
+    if (!username) {
+        return <>We forgot your username... awkward.</>;
+    }
+
+    const onSubmit = async (request: ConfirmUserRequest) => {
+        try {
+            await confirmUser(request);
+            router.push('/auth/login');
+            return;
+        } catch (error: unknown) {
+            if (isAWSError(error, 'CodeMismatchException')) {
```

Comment:
I think we can just check if it's an instance of CodeMismatchException, but I'm not sure if these are provided for TypeScript.